core/src/main/java/org/mule/runtime/core/internal/policy/PolicyNextActionMessageProcessor.java

```diff
@@ -103,30 +108,48 @@ public CoreEvent process(CoreEvent event) throws MuleException {
   private OnExecuteNextErrorConsumer errorConsumer(PolicyEventMapper policyEventMapper,
                                                    PolicyNotificationHelper notificationHelper) {
 
-    if (isWithinSourcePolicy(getLocation())) {
```

Comment:
if you remove this pre-check you will add complexity to every error handling, affecting perfromance